llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8539,6 +8539,20 @@ VPRecipeBuilder::tryToCreateWidenRecipe(Instruction *Instr,
   return tryToWiden(Instr, Operands, VPBB);
 }
 
+// EVL transform doesn't support backends where EVL diffs from RuntimeVF
+// in the second-to-last iteration.
+// Return false if the vector region has recipes relying on
+// RuntimeVF.
+static bool isCompatibleToEVLTransform(VPlan &Plan) {
+  if (any_of(Plan.getVectorLoopRegion()->getEntryBasicBlock()->phis(),
+             [](VPRecipeBase &Phi) {
+               return (isa<VPWidenIntOrFpInductionRecipe>(&Phi) ||
+                       isa<VPWidenPointerInductionRecipe>(&Phi));
+             }))
+    return false;
```

Comment:
nit: can replace `if()` with direct `return` (possibly needing to negate/or use `all_of`. With it being that compact, it would probably be better to directly fold this into the transform that introduces EVL

Reply:
OK, let's try this way.